tests/bluetooth/host/buf/bt_buf_get_cmd_complete/src/main.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <ztest.h>
+#include "kconfig.h"
+#include "hooks.h"
+#include "buf_help_utils.h"
+
+#include <zephyr/bluetooth/hci.h>
+#include <host/hci_core.h>
+#include <zephyr/bluetooth/buf.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/sys/util.h>
+
+/*
+ *  Return value from bt_buf_get_cmd_complete() should be NULL
+ *
+ *  This is to test the behaviour when memory allocation request fails
+ *
+ *  Constrains:
+ *     - bt_dev.sent_cmd value is NULL
+ *     - Timeout value is a positive non-zero value
+ *     - net_buf_alloc_fixed() returns a NULL value
+ *
+ *  Expected behaviour:
+ *     - net_buf_alloc_fixed() to be called with the correct memory allocation pool
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns NULL
+ *
+ */
+void test_returns_null_sent_cmd_is_null(void)
+{
+	struct net_buf *buf;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
+
+	ztest_returns_value(net_buf_alloc_fixed, NULL);
+	ztest_expect_value(hooks_net_buf_alloc_fixed_timeout_validation_hook, value, timeout.ticks);
+	buf = bt_buf_get_cmd_complete(timeout);
+	zassert_is_null(buf, "Return value was not NULL");
+}
+
+/*
+ *  Return value from bt_buf_get_cmd_complete() shouldn't be NULL
+ *
+ *  Constrains:
+ *     - bt_dev.sent_cmd value is NULL
+ *     - Timeout value is a positive non-zero value
+ *     - net_buf_alloc_fixed() return a not NULL value
+ *
+ *  Expected behaviour:
+ *     - net_buf_alloc_fixed() to be called with the correct memory allocation pool
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns the same value returned by net_buf_alloc_fixed()
+ *     - Return buffer type is set to BT_BUF_EVT
+ *
+ */
+void test_returns_not_null_sent_cmd_is_null(void)
+{
+	static struct net_buf test_reference;
+	struct net_buf *buf;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
+
+	ztest_expect_value(net_buf_simple_reserve, buf, &test_reference.b);
```

Comment:
In my opinion, if we are going to override an inline function, I'll vote for editing the current header file to allow overriding.

Creating a new header file will require copy the original file content and modify the declaration of the inline functions only.
That will create some sort of inconsistency and it will be harder to maintain the copies compatible over time.